src/android/InAppBrowser.java

```diff
@@ -945,23 +855,96 @@ public boolean onKey(View v, int keyCode, KeyEvent event) {
                         }
                         mUploadCallbackLollipop = filePathCallback;
 
+                        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+
+                        if(takePictureIntent.resolveActivity(cordova.getActivity().getPackageManager()) != null) {
+
+                            File photoFile = null;
+                            Uri photoUri = null;
+
+                            if (isAndroidQ) {
+                                // Android Q compatibility
+                                photoUri = createImageUri();
+                                mCameraUri = photoUri;
+                                if (photoUri != null) {
+                                    takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, photoUri);
+                                }
+                            } else {
+                                try {
+                                    photoFile = createImageFile();
+                                    takePictureIntent.putExtra("PhotoPath", mCM);
+                                } catch (IOException ex) {
+                                    Log.e(LOG_TAG, "Image file creation failed", ex);
+                                }
+                                if (photoFile != null) {
+                                    mCM = "file:" + photoFile.getAbsolutePath();
+                                    takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, Uri.fromFile(photoFile));
+                                } else {
+                                    takePictureIntent = null;
+                                }
+                            }
+                        }
                         // Create File Chooser Intent
-                        Intent content = new Intent(Intent.ACTION_GET_CONTENT);
-                        content.addCategory(Intent.CATEGORY_OPENABLE);
-                        content.setType("*/*");
+                        Intent contentSelectionIntent = new Intent(Intent.ACTION_GET_CONTENT);
+                        contentSelectionIntent.addCategory(Intent.CATEGORY_OPENABLE);
+                        contentSelectionIntent.setType("*/*");
+                        Intent[] intentArray;
+                        if(takePictureIntent != null){
+                            intentArray = new Intent[]{takePictureIntent};
+                        }else{
+                            intentArray = new Intent[0];
+                        }
+
+                        Locale localeInfo = cordova.getActivity().getApplicationContext().getResources()
+                                .getConfiguration().getLocales().get(0);
+
+                        String lang = localeInfo.toLanguageTag();
+
+                        String title;
+
+                        if (lang.startsWith("en"))
+                        {
+                            title = "Choose the source";
```

Comment:
Adding hard-coded strings such as this is undesirable - better to let the plugin user pass in a string in their locale